### PR TITLE
Feat/adp 2802 implement serialization of tx main body

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

We want to remove the dependency of the SDK with the CML library. For this we need to develop our own serialization logic around the Cardano domain types.

# Proposed Solution

- Implement a set of CBOR classes that follow the Reader/Writer pattern. These classes will allow us to be very precise and control how the types are serialized. This will also allow us to solve the round trip problem we currently have due difference on transactions generated by the CLI or the CSL/CML library.

- Create a Transaction class which can serialize and serialize to CBOR format. This class must also be able to convert to and from Core.Tx type

# Important Changes Introduced

- Removed the dependency on borc library.
- CborReader and CborWriter introduced.

### CODE REVIEW

File: packages/cardano-services/src/TxSubmit/rabbitmq/TxSubmitWorker.ts [hunk 1]
```diff
@@ -294,7 +294,7 @@ export class TxSubmitWorker extends EventEmitter {
       const txBody = new Uint8Array(content);
 
       // Register the handling of current transaction
-      txId = cmlUtil.deserializeTx(txBody).id;
+      txId = (await cmlUtil.deserializeTx(txBody)).id;
```
Comment (mkazlauskas) [inline packages/cardano-services/src/TxSubmit/rabbitmq/TxSubmitWorker.ts:297]: Just noticed that we have a more efficient way to do this:
```suggestion
      txId = TransactionId.fromTxBodyCbor(TxBodyCBOR.fromTxCBOR(txBody))
```
Maybe we can also update those utils to (at least partially) use new serialization?

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

**Review by mkazlauskas — DISMISSED** [review]
Great work @AngelCastilloB! :rocket:

**Comment by mirceahasegan** [comment]
This commit is not verified: [64513af](https://github.com/input-output-hk/cardano-js-sdk/pull/693/commits/64513af48c8a30e055c6fe83a45babf69a10532f)

**Review by mirceahasegan — APPROVED** [review]
Fantastic work, @AngelCastilloB :mage_man:

**Review by mkazlauskas — APPROVED** [review]
:rocket:

**Review by rhyslbw — APPROVED** [review]
Great work @AngelCastilloB :mechanical_arm:

**Pull request merged**